src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -531,7 +546,19 @@ const WorkflowView = _.flow(
             ]),
             span({ style: { color: colors.darkBlue[0], fontSize: 24 } }, name)
           ]),
-          div({ style: { marginTop: '0.5rem' } }, `Snapshot ${methodVersion}`),
+          div({ style: { marginTop: '0.5rem' } },
+            (sourceRepo === 'agora') ? [
+              'Snapshot', div({ style: { display: 'inline-block', marginLeft: '0.25rem', minWidth: 60  } }, [
+                h(Select, {
+                  isClearable: false,
+                  isSearchable: false,
+                  value: methodVersion,
+                  getOptionLabel: ({ value }) => Utils.normalizeLabel(value),
+                  options: snapshotIds,
+                  onChange: chosenSnapshot => this.loadNewMethodConfig(chosenSnapshot.value)
```

Comment:
Is this being triggered on the initial render? I'm seeing the select having no value; looks like it's loading. However, I think on initial render, we have all of the information we need.

Reply:
moved the set state to the component did mount, which removes the extra delay on rendering the select